### ISSUE TITLE
How to visualize attention map in figure 4

### ISSUE THREAD

**Issue by Pixie8888** [facebookresearch/noisy_few_shot#8]
Hi,

Thanks for releasing code! May I know how to plot the attention map in figure 4?

**Comment by kevinjliang** [comment]
I unfortunately don’t have simple code for doing this. To get the attention maps, I made a copy of the PyTorch implementation of multi-headed attention (mha), to output the attention weights without averaging across heads. I also created a copy of the transformer encoder layer with this modified copy of mha substituted in, and then used that in my model. I used this modified implementation to forward pass on some data, and inspected the intermediate results. The result is a tensor that is H x N*(K+1) x N*(K+1), where H is the number of heads; each of H maps should be plotted individually.

Since then, PyTorch has added the ability for the mha functional to output the attention masks unaveraged: https://github.com/pytorch/pytorch/blob/main/torch/nn/functional.py#L5125. As result, instead of making a copy of mha, you can probably just add the appropriate flags.
